algorithm/recursive/permutation

```diff
@@ -0,0 +1,32 @@
+#include <iostream>
+#include <string>
+using namespace std;
+
+// list all rearrangement of string n of distinct characters
+
+// pos is recursive depth 
+void perm(string str, bool visited[], int pos, string ans){
+    if(pos <= 0){
+        cout << ans << " ";
+        return;
+    }
+    for(int i = 0 ; i < str.size(); i++){
+        if(!visited[i]){
+            visited[i] = true;
+            perm(str, visited, pos-1, ans+str[i]);
+            visited[i] = false; 
+        }
+    }
+}
+
+int main(){
+    string str, ans = "";
+    cin >> str;
+    bool visited [str.size()];
```

Comment:
Although C++ support dynamic array length declaration.
Try not to use it so often.
Use STL vector instread, or use static length declaration.